processor/src/fast/basic_block.rs

```diff
@@ -45,7 +45,7 @@ impl FastProcessor {
 
         let mut batch_offset_in_block = 0;
         let mut op_batches = basic_block_node.op_batches().iter();
-        let mut decorator_ids = basic_block_node.decorator_iter();
+        let mut decorator_ids = basic_block_node.indexed_decorator_iter();
```

Comment:
Should we remove the call to `self.execute_before_enter_decorators(node_id, program, host)?;` on line 41? Otherwise IIUC this would re-execute the `before_enter` decorators before the first operation in the block.

Reply:
This specific iterator does not include `before_enter` (or `after_exit` decorators).

Reply:
I've included the processor tests. Note the behavior of `indexed_decorator_iter` was also covered by `test_indexed_decorator_iter_excludes_before_enter_after_exit`.

I've made the doc comments on all iterator-returning methods more clear as well re: what they do or do not include.